tools/GenerateCmdletDesignMarkdown.ps1

```diff
@@ -0,0 +1,121 @@
+    Write-Debug "Get all cmdlets md file under the $Path folder."
+    $cmdLets = Get-ChildItem -Path $Path -Filter '*-*.md' | Select-Object -Property FullName, Name
+
+    # Add Cmdlet, Verb, Noun property for sort object.
+    $cmdLets | Add-Member -NotePropertyName Cmdlet -NotePropertyValue $null
+    $cmdLets | Add-Member -NotePropertyName Verb -NotePropertyValue $null
+    $cmdLets | Add-Member -NotePropertyName Noun -NotePropertyValue $null
+
+    # set priority for the specified cmdlets.
+    if ($PSBoundParameters.ContainsKey("NounPriority")) {
+        $priority = 0
+        $NounPriorityHash = @{}
+        foreach ($cmdlet in $NounPriority) {
+            $NounPriorityHash.Add($cmdlet, $priority++)
+        }
+    }
+
+    for ($index = 0; $index -lt $cmdLets.Length; $index++) {
+        # Join 0 prefix with New verb.
+        $verb = $cmdLets[$index].Name.Split("-")[0] -eq 'New' ? '0New' : $cmdLets[$index].Name.Split("-")[0]
+        # Join priority with Noun.
+        $originNoun = $cmdLets[$index].Name.Split("-")[1].Split(".")[0];
+        $Noun = $null -eq $NounPriorityHash ? $originNoun : ($null -eq $NounPriorityHash[$originNoun] ? $originNoun : $NounPriorityHash[$originNoun].ToString() + $originNoun)
+    
+        $cmdLets[$index].Cmdlet = $cmdLets[$index].Name.Split(".")[0];
+        $cmdLets[$index].Verb = $verb
+        $cmdLets[$index].Noun = $Noun
+    }
+
+
+    $sortedCmdlets = $cmdLets | Sort-Object -Property @{Expression = "Noun"; Descending = $false }, @{Expression = "Verb"; Descending = $false }
+
+    "The sorted cmdles list : " | Write-Debug
+    $sortedCmdlets | Out-String | Write-Debug
+
+
+    $outFilePath = Join-Path $OutPath $OutputFileName
+
+    # Try remove output file
+    Write-Debug "Delete the $OutputFileName file if it exists."
+    Remove-Item -Path $outFilePath -ErrorAction SilentlyContinue
+
+
+    # Get cmdlet design message from the Verb-Noun.md file.
+    foreach ($cmdlet in $sortedCmdlets) {
+        $content = Get-Content -Path $cmdlet.FullName -ErrorAction Stop
+        $contentStr = ($content | Out-String)
+        $designDoc = $contentStr.Substring($contentStr.IndexOf("# $($cmdlet.Cmdlet)"), $contentStr.IndexOf('## PARAMETERS') - $contentStr.IndexOf("# $($cmdlet.Cmdlet)"))
+    
+        if ($designDoc.Contains("{{ Add title here }}")) {
+            $designDoc = $designDoc.Remove($designDoc.IndexOf('## DESCRIPTION'))
+        }
+        else {
+            $designDoc = $designDoc.Remove($designDoc.IndexOf('## DESCRIPTION'), $designDoc.IndexOf('## EXAMPLES') - $designDoc.IndexOf('## DESCRIPTION'))
+        }
+
+        $designDoc = $designDoc -replace '```(\r\n\w{1})', '```powershell$1'
+        $designDoc = $designDoc -replace '###', '+'
+        $designDoc = $designDoc -replace '#+', '####'
+
+        $designDoc | Out-File -FilePath $outFilePath -Append -ErrorAction Stop
+    }
+
+    Write-Host "Genereated $outFilePath completed."
+    return
+}
+catch {
+    throw
+    return
```

Comment:
Remove this line because it is inaccessible

Reply:
Done